### PR TITLE
[WIP] pyBeam integration into SU2 as a subproject

### PR DESCRIPTION
**Pull request by rsanfer**
## Proposed Changes
This PR integrates pyBeam, a non-linear open source beam solver, into the SU2 infrastructure as a subproject, and adds support for coupled FSI simulations using SU2's flow solver and pyBeam's structural solver.

Tutorials on how to use the implementation will come soon. At this point, only minor changes to the compilation and the python scripts are required. The impact to the core of the code is minor.

I will be away for a few days, but please go ahead and have a look/comment/find room for improvement in the meantime :) I will reply to your comments as soon as I'm back to office.

I also added a couple of test cases, but I haven't yet found the best way to hook them to our regression test system. I'll definitely look into it before the PR is merged, but due to personal time constraints, it'd be great if this could be integrated for 7.0.3.

## Related Work
This work is the result of a collaboration with @RoccoBombardieri and @crauno from the University Carlos III Madrid. 

Please refer to a recent [paper at EUROGEN 2019](https://www.researchgate.net/publication/335972259_Towards_an_open-source_framework_for_aero-structural_design_and_optimization_within_the_SU2_suite) with regards to this work. Full reference will be provided here upon publication in an Springer ECCOMAS book series.

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [x] I am submitting my contribution to the develop branch.
- [x] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags).
- [x] My contribution is commented and consistent with SU2 style.
- [ ] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp) , if necessary.

### CODE REVIEW

File: SU2_PY/SU2_FSI/AdjointInterface.py [hunk 1]
```diff
@@ -0,0 +1,866 @@
...
+#                 Prof. Nicolas R. Gauger's group at Kaiserslautern University of Technology.
+#                 Prof. Alberto Guardone's group at Polytechnic University of Milan.
+#                 Prof. Rafael Palacios' group at Imperial College London.
+#                 Prof. Edwin van der Weide's group at the University of Twente.
+#                 Prof. Vincent Terrapon's group at the University of Liege.
+#
+# Copyright (C) 2012-2019 SU2, the open-source CFD code.
+#
+# SU2 is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
```
Comment (pcarruscag) [inline SU2_PY/SU2_FSI/AdjointInterface.py]: This does not match what we have in other files also, 7.0.2 (same for other files)

File: TestCases/pybeam_fsi/primal/configFlow.cfg [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+% -------------------------- MULTIGRID PARAMETERS -----------------------------%
+
+MGLEVEL= 0
+MGCYCLE= V_CYCLE
+
+MG_PRE_SMOOTH= ( 1, 2, 3, 3 )
+MG_POST_SMOOTH= ( 0, 0, 0, 0 )
+MG_CORRECTION_SMOOTH= ( 0, 0, 0, 0 )
+MG_DAMP_RESTRICTION= 0.7
+MG_DAMP_PROLONGATION= 0.7
```
Comment (pcarruscag) [inline TestCases/pybeam_fsi/primal/configFlow.cfg]: please consider removing irrelevant options

**Comment by RoccoBombardieri** [comment]
WIP

**Comment by rsanfer** [comment]
Hi guys,
this has been sitting here for some time, and I just noticed there are some conflicts, etc. I'm very sorry that I haven't been able to update it, and I can't promise I will be able to look into it soon enough, but maybe @RoccoBombardieri could have a look into it? :)

**Comment by RoccoBombardieri** [comment]
Hello, guys, will give a look ASAP! I see the conflicting files are not the ones we are modifying or anything in the current PR. I'll double check the source of the problem.

**Comment by RoccoBombardieri** [comment]
WIP

**Comment by RoccoBombardieri** [comment]
WIP2

**Comment by RoccoBombardieri** [comment]
Hello @pcarruscag,

I've been following the conversation about #1124 and the possible conflicts with this PR. I wanted to say that, with respect to this specific PR I kept working on a different branch lately (**feature_pyBeam_ShapeDesignV2**) which, is currently the most advanced version of the project. How to conciliate? Shall we pull from feature_pyBeam_ShapeDesignV2 directly here? 

One issue of that version is that I had to introduce some changes into the CFD solver to extend to FSI the fixed Cl design feature. I thought the mods where general but that actually broke something within SU2 CFD AD. 

Best,
Rocco

**Comment by pcarruscag** [comment]
There is no conflict because both here and there the files were moved, to SU2_FSI and FSI_tools respectively.
What there is now is a massive amount of duplication, if you and Nicola want to try and fix that, it would be the ideal outcome for SU2.

I made the final changes Tom and I had requested on this PR because it did not seem like this was going anywhere... But if you are still working on it, well do what you think is best, I don't have time to figure out how to get rid of the duplication.

"develop" needs to work as a staging area, if you have a chunk of work that can be merged, without breaking compilation, testcases, or interfering too much with the work of other folks, then it should be merged ASAP to avoid this kind of situation.
For example refactoring this "FSIInterface.py" file into direct and adjoint should have possibly been a PR by itself to set the stage for the rest.

Anyway, regarding the initial proposed changes this looks good to me.

**Review by pcarruscag — APPROVED** [review]

**Pull request closed without merging**